src/dmi.rs

```diff
@@ -26,24 +26,41 @@ byond_fn!(fn dmi_resize_png(path, width, height, resizetype) {
 });
 
 fn strip_metadata(path: &str) -> Result<()> {
-    let (info, image) = read_png(path)?;
-    write_png(path, info, image)
+    let (reader, frame_info, image) = read_png(path)?;
+    write_png(path, reader, frame_info, image, true)
 }
 
-fn read_png(path: &str) -> Result<(OutputInfo, Vec<u8>)> {
+fn read_png(path: &str) -> Result<(Reader<File>, OutputInfo, Vec<u8>)> {
     let mut reader = Decoder::new(File::open(path)?).read_info()?;
     let mut buf = vec![0; reader.output_buffer_size()];
+    let frame_info = reader.next_frame(&mut buf)?;
 
-    let out_info = reader.next_frame(&mut buf)?;
-    Ok((out_info, buf))
+    Ok((reader, frame_info, buf))
 }
 
-fn write_png(path: &str, info: OutputInfo, image: Vec<u8>) -> Result<()> {
+fn write_png(
+    path: &str,
+    reader: Reader<File>,
+    info: OutputInfo,
+    image: Vec<u8>,
+    strip: bool,
```

Comment:
Doesn't look like you set this to false anywhere. Needed?

Reply:
> Also threw in a zTxt copy mechanism if at some point we want it, because we now have access to that information there and it's possible in the new library version.

Reply:
also this fix works, will merge tomorrow and cut release
https://github.com/ZeWaka/tgstation/pull/2 (ignore the dumb tramstation random error)

Reply:
aka, useful for preserving DMI information

Reply:
Oh okay